static/js/containers/CourseSearchPage.js

```diff
@@ -222,12 +219,15 @@ export class CourseSearchPage extends React.Component<Props, State> {
     }
     // Retain the current facet options and counts for this facet group
     const updatedFacetGroup =
-      currentFacetGroup && currentFacetGroup.get(name)
+      currentFacetGroup && currentFacetGroup.group === name
         ? currentFacetGroup
-        : new Map([[name, facets.get(name)]])
+        : {
+          group:  name,
+          result: facets.get(name)
+        }
+    // $FlowFixMe: nothing undefined here
```

Comment:
I think there is an issue in the logic here. In particular, if we have the case that `currentFacetGroup.group === name` I don't think we want to just set `updatedFacetGroup` to be `currentFacetGroup`. It is possible, for instance, that there are new options for the facet present in `facets.get(name)` which aren't saved in `currentFacetGroup` it leads to the very behavior that this PR is trying to eliminate. Here's how I can reproduce the issue:

1. start with a full page refresh on `/courses/search`
2. select 'OCW' for platform
3. select 'Available Now' for 'availability'
4. de-select 'OCW' for platform'
5. de-select 'Available Now' for 'availability

From this point if I select and de-select 'Available Now' the options shown in the facet change - if I have nothing selected I see all the options, but selecting 'Available Now' causes the other options to go away so that I only see the selected 'Available Now' option.

Here's a gif that shows the issue:

![issue](https://user-images.githubusercontent.com/6207644/54760040-a297ab00-4bc5-11e9-9af4-53468dcc6e9a.gif)


This is happening because, in the above scenario, when I de-select 'Available Now' the `currentFacetGroup` is set to be the 'availability' facet with just the 'available now' option present. Then the search is run (fired by the `componentDidUpdate` hook) and more facet values come back in the response, since at this point I have no facets selected and all values are returned. This means that in the `mergeFacetOptions` method when we call 

```js
mergeFacetResults(currentFacetGroup.result, emptyActiveFacets, facets.get(group))
```

we're going to get all options displaying in the UI because they're all present in `facets.get(group)`. But then, if I re-select 'Available Now' then `currentFacetGroup` isn't updated, because `currentFacetGroup.group === name` when this `toggleFacet` method is called. So then when we call `mergeFacetOptions` to render the `availability` facet we only get the 'Available Now' option in the UI, because when we make the above call to `mergeFacetOptions` we have got only the 'Available Now' option in both `currentFacetGroup.result` _and_ in `facets.get(group)`. 

Does that make sense?

I believe this can be address by changing the `toggleFacet` method to merge `facets.get(name)` with `currentFacetGroup.results` when `currentFacetGroup.group === name`, something like this:


```js
this.setState({
  activeFacets:      updatedFacets,
  currentFacetGroup: {
    group:  name,
    result:
      currentFacetGroup && currentFacetGroup.group === name
        ? mergeFacetResults(currentFacetGroup.result, facets.get(name))
        : facets.get(name)
  }
})
```

Reply:
Thanks, updated